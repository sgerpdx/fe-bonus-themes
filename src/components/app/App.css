@font-face {
  font-family: 'Roddenberry';
  src: local('Roddenberry'),
    url('../../fonts/Roddenberry.otf') format('opentype');
  font-weight: normal;
}

body {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  font-family: 'Roddenberry';
  text-align: center;
}

.container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.displayArea {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.displayAreaVulcan {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  background-color: rgb(245, 208, 153);
  color: maroon;
}

.displayAreaSpace {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  background-color: black;
  color: white;
}

.displayAreaEarth {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  background-color: lightblue;
  color: rgb(141, 107, 20);
}

.characterGrid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  list-style-type: none;
}

.headerThemeSpace {
  background-color: black;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: left;
  padding-left: 20px;
  font-family: Arial, Helvetica, sans-serif;
}

.headerThemeEarth {
  background-color: lightblue;
  color: rgb(141, 107, 20);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: left;
  padding-left: 20px;
  font-family: Arial, Helvetica, sans-serif;
}

.headerThemeVulcan {
  background-color: rgb(245, 208, 153);
  color: maroon;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: left;
  padding-left: 20px;
  font-family: Arial, Helvetica, sans-serif;
}
