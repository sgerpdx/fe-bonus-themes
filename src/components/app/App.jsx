import React, { useState } from 'react';
//import { Route, BrowserRouter as Router } from 'react-router-dom';
import ListContainer from '../../containers/ListContainer';
import Header from '../../containers/Header';

export default function App() {
  // const [theme, setTheme] = useState('space');

  // const handleThemeChange = (e) => {
  //   setTheme((theme) => (theme = e.target.value));
  // };

  return (
    <>
      <h1>Space: The Final Frontier...</h1>
      <Header />
      <ListContainer />
    </>
  );
}
