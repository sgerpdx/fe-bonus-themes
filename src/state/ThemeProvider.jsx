import React, { createContext, useState, useContext } from 'react';
//import PropTypes from 'prop-types';

export const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('space');

  const handleThemeChange = (e) => {
    setTheme((theme) => (theme = e.target.value));
  };

  const state = { theme, handleThemeChange };

  return (
    <ThemeContext.Provider value={state}>{children}</ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const { theme } = useContext(ThemeContext);
  return theme;
};

export const useThemeChange = () => {
  const { handleThemeChange } = useContext(ThemeContext);
  return handleThemeChange;
};

export default ThemeProvider;
