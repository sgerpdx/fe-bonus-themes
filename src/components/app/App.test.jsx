/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import ThemeProvider from '../../state/ThemeProvider';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
