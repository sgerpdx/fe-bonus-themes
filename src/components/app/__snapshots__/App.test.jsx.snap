// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <h1>
    Starfleet API Databank
  </h1>
  <div
    class="headerThemeSpace"
  >
    <section>
      <div>
        <h4>
          Use the menu below to select a visual theme:
        </h4>
        <select
          id="themes"
          name="themes"
        >
          <option
            value="space"
          >
            Space
          </option>
          <option
            value="earth"
          >
            Earth
          </option>
          <option
            value="vulcan"
          >
            Vulcan
          </option>
        </select>
      </div>
      <figure>
        <img
          height="40px"
          src="https://i.stack.imgur.com/mVOSg.gif"
        />
      </figure>
    </section>
  </div>
  <h2>
    Loading...
  </h2>
</DocumentFragment>
`;
